.github/workflows/check-changelog.yml

```diff
@@ -0,0 +1,45 @@
+name: changelog
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+
+jobs:
+  check:
+    name: Check changelog entry
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/github-script@v6
+        id: check
+        with:
+          script: |
+            const fs = require('node:fs')
+            const path = require('node:path')
+          
+            const { data: { labels: labels } } = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            })
+            const labelNames = labels.map(label => label.name)
+
+            if (labelNames.includes('skip-changelog')) {
+              return
+            }
+
+            let found = false
+            const changeTypes = ['packaging', 'added', 'changed', 'removed', 'fixed']
+            for (changeType of changeTypes) {
+              const filename = path.join('newsfragments', `${context.issue.number}.${changeType}.md`)
+              if (fs.existsSync(filename)) {
+                found = true
+                break
+              }
+            }
+          
+            if (!found) {
+              const errorMsg = '📝 Changelog entry not found, please add one to `newsfragments` folder.'
```

Comment:
TODO: add a link to https://github.com/PyO3/pyo3/blob/main/Contributing.md#documenting-changes